kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/B2BPurchaseOrderSciquestServiceImpl.java

```diff
@@ -0,0 +1,375 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.module.purap.PurapParameterConstants;
+import org.kuali.kfs.module.purap.businessobject.PurApItem;
+import org.kuali.kfs.module.purap.businessobject.PurchaseOrderItem;
+import org.kuali.kfs.module.purap.document.PurchaseOrderDocument;
+import org.kuali.kfs.module.purap.util.PurApDateFormatUtils;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.sys.service.impl.KfsParameterConstants;
+import org.kuali.kfs.vnd.businessobject.ContractManager;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.kim.api.identity.Person;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.arizona.kfs.module.purap.PurapConstants;
+import edu.arizona.kfs.sys.KFSConstants;
+
+@SuppressWarnings("unused")
+public class B2BPurchaseOrderSciquestServiceImpl extends org.kuali.kfs.module.purap.document.service.impl.B2BPurchaseOrderSciquestServiceImpl {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(B2BPurchaseOrderSciquestServiceImpl.class);
+
+    private ConfigurationService configurationService;
+    private DateTimeService dateTimeService;
+
+    private DateTimeService getDateTimeService() {
+        if (dateTimeService == null) {
+            dateTimeService = SpringContext.getBean(DateTimeService.class);
+        }
+        return dateTimeService;
+    }
+
+    private ConfigurationService getConfigurationService() {
+        if (configurationService == null) {
+            configurationService = SpringContext.getBean(ConfigurationService.class);
+        }
+        return configurationService;
+    }
+
+    public B2BPurchaseOrderSciquestServiceImpl() {
+        super();
+    }
+
+    @Override
+    public String getCxml(PurchaseOrderDocument purchaseOrder, String requisitionInitiatorId, String password, ContractManager contractManager, String contractManagerEmail, String vendorDuns) {
+        Person initiator = getPersonService().getPersonByPrincipalName(requisitionInitiatorId);
+        StringBuilder cxml = new StringBuilder(5000);
+        cxml = writeHeader(cxml, password);
+        cxml.append("  <PurchaseOrder>\n");
+        cxml = writePOHeader(cxml, purchaseOrder, initiator, contractManagerEmail, contractManager, vendorDuns);
+        cxml = writePOLines(cxml, purchaseOrder);
+        cxml.append("  </PurchaseOrder>\n");
+        cxml.append("</PurchaseOrderMessage>");
+        LOG.debug("getCxml(): cXML for po number " + purchaseOrder.getPurapDocumentIdentifier() + ":\n" + cxml.toString());
+        return cxml.toString();
+    }
+
+    protected StringBuilder writeHeader(StringBuilder cXml, String password) {

```

Comment:
If we copy what was in 6 and change the one line in the getCxml(), this is not needed.